worlds/ror2/Rules.py

```diff
@@ -7,27 +7,37 @@ class RiskOfRainLogic(LogicMixin):
     def _ror_has_items(self, player: int, amount: int) -> bool:
         count: int = self.item_count("Common Item", player) + self.item_count("Uncommon Item", player) + \
                      self.item_count("Legendary Item", player) + self.item_count("Boss Item", player) + \
-                     self.item_count("Lunar Item", player) + self.item_count("Equipment", player)
+                     self.item_count("Lunar Item", player) + self.item_count("Equipment", player) + \
+                     self.item_count("Dio's Best Friend", player)
         return count >= amount
 
 
 def set_rules(world: MultiWorld, player: int):
-    total_checks = world.total_locations[player]
     # divide by 5 since 5 levels (then commencement)
-    items_per_level = total_checks / 5
-    leftover = total_checks % 5
+    items_per_level = max(int(world.total_locations[player] / 5 / (world.item_pickup_step[player]+1)), 1)
```

Comment:
Why divide by item_pickup_step? I wonder what the logic here is, I might be missing it.

Reply:
I'm dividing by item_pickup_step because as that value is increased the rate at which the player is sending items to the server is decreased significantly so this way the playthrough doesn't load as many items into early spheres for those players with a higher item_pickup_step. This can end up with a very large amount of the RoR2 player's item pool being almost exclusively junk but in my opinion is better than locking someone's important sphere 1 item behind 40 chest opens.